src/NuGet.Core/NuGet.Commands/RestoreCommand/Utility/MSBuildRestoreUtility.cs

```diff
@@ -30,9 +30,19 @@ public static class MSBuildRestoreUtility
         private const string DoubleSlash = "//";
 
         /// <summary>
-        /// Convert MSBuild items to a DependencyGraphSpec.
+        /// Creates a <see cref="DependencyGraphSpec" /> from the specified MSBuild items.
         /// </summary>
+        /// <param name="items">An <see cref="IEnumerable{T}" /> of <see cref="IMSBuildItem" /> objects representing the MSBuild items gathered for restore.</param>
         public static DependencyGraphSpec GetDependencySpec(IEnumerable<IMSBuildItem> items)
```

Comment:
Any reason not to make this the default?

It's only used in CLI scenarios right?
It's VS where mutation is a concern correct?

Reply:
It's not clear to me why VS would need to mutate a DGSpec. If projects are loaded/unloaded, a new DGSpec can be created with the new project list.

Reply:
I would also like a better understanding of why the items in the DG spec are cloned?

Reply:
The readonly part can be a bit misleading.
Readonly means that the PackageSpec is not cloned. 

The primary reason around package spec cloning is defensiveness. In Visual Studio as a long lived process, the PackageSpec is not guaranteed to be immutable. 

After this, we're down to 2 clones, 1 in the RestoreRunner, 1 in the CPSPackageReferenceProject, the latter one is needed, but the previous I'm 99% sure we don't need anymore. 

Long term the approach should be; 

1. Introduce a PackageSpec lock mechanism, and the dg spec would only accept locked PackageSpec. Then it can just avoid cloning.